### PR TITLE
Storefront country picker

### PR DESCRIPTION
**Pull request by arudnicka**
- [ ] Missing rspecs

### CODE REVIEW

File: frontend/app/views/spree/shared/_link_to_stores.erb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<% stores.each do |store| %>
+  <div class="pl-3 d-flex justify-content-between align-items-center">
+    <span class="flag-icon flag-icon-<%= store.default_country.iso.downcase %>"></span>
+    <%= link_to store.default_country.name, store.url, class: 'dropdown-item text-uppercase' %>
```
Comment (damianlegawiec) [inline frontend/app/views/spree/shared/_link_to_stores.erb:4]: @arudnicka there's no currency here

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 2]
```diff
@@ -7,6 +7,10 @@ def body_class
       @body_class
     end
 
+    def store_country
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb:10]:
```suggestion
    def store_country_iso(store)
      store ||= current_store
```

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 3]
```diff
@@ -7,6 +7,10 @@ def body_class
...
+      current_store.default_country.iso.downcase
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb:11]:
```suggestion
      return unless store
      return unless store.default_country
      
      store.default_country.iso
```

File: frontend/app/views/spree/shared/_link_to_stores.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<% stores.each do |store| %>
+  <div class="pl-3 d-flex justify-content-between align-items-center">
+    <span class="flag-icon flag-icon-<%= store.default_country.iso.downcase %>"></span>
```
Comment (damianlegawiec) [inline frontend/app/views/spree/shared/_link_to_stores.erb:3]:
```suggestion
    <span class="flag-icon flag-icon-<%= store_country_iso(store) %>"></span>
```

**Review by AleksandraKaminska — APPROVED** [review]

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 5]
```diff
@@ -7,6 +7,22 @@ def body_class
...
+    def store_country_iso(store)
+      store ||= current_store
+      return unless store
+      return unless store.default_country
+
+      store.default_country.iso.downcase
+    end
+
+    def stores
+      @stores ||= Spree::Store.all.eager_load(:default_country)
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb:19]:
```suggestion
      @stores ||= Spree::Store.include(:default_country)
```

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 6]
```diff
@@ -7,6 +7,22 @@ def body_class
...
+    end
+
+    def store_currency_symbol(currency)
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb:22]:
```suggestion
    def store_currency_symbol(currency)
      return if currency&.blank?
      
```

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 7]
```diff
@@ -7,6 +7,22 @@ def body_class
       @body_class
     end
 
+    def store_country_iso(store)
+      store ||= current_store
+      return unless store
+      return unless store.default_country
+
+      store.default_country.iso.downcase
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb:15]:
```suggestion
      store.default_country&.iso&.downcase
```

File: core/db/migrate/20200423123001_add_default_country_id_to_spree_store.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+class AddDefaultCountryIdToSpreeStore < ActiveRecord::Migration[6.0]
+  def change
+    unless column_exists?(:spree_stores, :default_country_id)
+      add_column :spree_stores, :default_country_id, :integer
```
Comment (damianlegawiec) [inline core/db/migrate/20200423123001_add_default_country_id_to_spree_store.rb:4]:
```suggestion
      add_column :spree_stores, :default_country_id, :integer
      Spree::Store.reset_column_information
      Spree::Store.update_all(default_country_id: Spree::Config[:default_country_id])
```

File: core/app/models/spree/store.rb [hunk 9]
```diff
@@ -2,6 +2,7 @@ module Spree
   class Store < Spree::Base
     has_many :orders, class_name: 'Spree::Order'
     has_many :payment_methods, class_name: 'Spree::PaymentMethod'
+    belongs_to :default_country, class_name: 'Spree::Country'
 
     with_options presence: true do
       validates :name, :url, :mail_from_address
```
Comment (damianlegawiec) [inline core/app/models/spree/store.rb:8]:
```suggestion
      validates :name, :url, :mail_from_address, :default_country_id, :default_currency
```

File: backend/app/views/spree/admin/stores/_form.html.erb [hunk 10]
```diff
@@ -88,6 +88,11 @@
       <%= f.select :default_locale, options_from_collection_for_select(all_locales_options, :last, :first, @store.default_locale) %>
       <%= f.error_message_on :default_locale %>
     <% end %>
+    <%= f.field_container :default_country_id, class: ['form-group'] do %>
+      <%= f.label :default_country_id, Spree.t('i18n.country') %>
+      <%= f.select :default_country_id, options_from_collection_for_select(Spree::Country.all, :id, :name, @store.default_country_id) %>
```
Comment (damianlegawiec) [inline backend/app/views/spree/admin/stores/_form.html.erb:93]:
```suggestion
      <%= f.select :default_country_id, options_from_collection_for_select(@countries :id, :name, @store.default_country_id) %>
```

Let's move this `Spree::Country.all` to a controller `before_action`

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**